### ISSUE TITLE
Face Recognition with Caffe

### ISSUE THREAD

**Issue by DasPoti** [BVLC/caffe#2571]
Hi everybody,

I'm getting some trouble while training a convolutional net in caffe with the purpose of recognize faces. 

I'm trying to train this net: 

name: "FacesCaffeNet"
layers {
  name: "data"
  type: IMAGE_DATA
  top: "data"
  top: "label"
  image_data_param {
    source: "/imatge/sdelgado/workspace/caffe/faceNet/trainingFRAVD.txt"
    batch_size: 128 
    shuffle: true
  }
  transform_param {
    crop_size: 47
    mirror: true
  }
  include: { phase: TRAIN }
}
layers {
  name: "data"
  type: IMAGE_DATA
  top: "data"
  top: "label"
  image_data_param {
    source: "/imatge/sdelgado/workspace/FRAV2D/val/valFRAVD.txt"
    batch_size: 2
  }
  transform_param {
    crop_size: 47
    mirror: true
  }
  include: { phase: TEST }
}
layers {
  name: "conv1"
  type: CONVOLUTION
  blobs_lr: 1
  blobs_lr: 1
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 20
    kernel_size: 4
    stride: 1

  }
  bottom: "data"
  top: "conv1"
}
layers {
  name: "relu1"
  type: RELU
  bottom: "conv1"
  top: "relu1"
}
layers {
  name: "pool1"
  type: POOLING
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
  bottom: "relu1"
  top: "pool1"
}
layers {
  name: "conv2"
  type: CONVOLUTION
  blobs_lr: 1
  blobs_lr: 1
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 40
    kernel_size: 3
    stride: 1

  }
  bottom: "pool1"
  top: "conv2"
}
layers {
  name: "relu2"
  type: RELU
  bottom: "conv2"
  top: "relu2"
}
layers {
  name: "pool2"
  type: POOLING
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
  bottom: "relu2"
  top: "pool2"
}
layers {
  name: "conv3"
  type: CONVOLUTION
  blobs_lr: 1
  blobs_lr: 1
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 60
    stride: 1
    kernel_size: 3

  }
  bottom: "pool2"
  top: "conv3"
}
layers {
  name: "relu3"
  type: RELU
  bottom: "conv3"
  top: "relu3"
}
layers {
  name: "conv4"
  type: CONVOLUTION
  blobs_lr: 1
  blobs_lr: 2
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 80
    kernel_size: 2
    stride: 1

  }
  bottom: "relu3"
  top: "conv4"
}
layers {
  name: "relu4"
  type: RELU
  bottom: "conv4"
  top: "relu4"
}

layers {
  name: "fc3"
  type: INNER_PRODUCT
  blobs_lr: 1
  blobs_lr: 2
  weight_decay: 1
  weight_decay: 0
  inner_product_param {
    num_output: 4096
    
  }
  bottom: "relu3"
  top: "fc3"
}

layers {
  name: "fc4"
  type: INNER_PRODUCT
  blobs_lr: 1
  blobs_lr: 2
  weight_decay: 1
  weight_decay: 0
  inner_product_param {
    num_output: 4096

  }
  bottom: "relu4"
  top: "fc4"
}

layers {
  name: "concat"
  bottom: "fc3"
  bottom: "fc4"
  top: "fcID"
  type: CONCAT
  concat_param {
    concat_dim: 1
  }
}

layers {
  name: "prob"
  type: SOFTMAX_LOSS
  bottom: "fcID"
  bottom: "label"
  top: "prob"
}

layers {
  name: "accuracy"
  type: ACCURACY
  bottom: "fcID"
  bottom: "label"
  top: "accuracy"
  include: { phase: TEST }
}

with the following parameters: 

test_iter: 50000
test_interval: 50000
base_lr: 0.01
lr_policy: "step"
gamma: 0.1
stepsize: 1000000
display: 100
max_iter: 400010
momentum: 0.9
weight_decay: 0.0005
snapshot: 10000

The loss does not decrease, at some point depending on the stepsize stops decreasing. The resulting loss value is around 5.8 or 6.2 dependeng on the stepsize.

Has anyone any idea about what to touch? Is there any stupid error on this net or something strange with the parameters? 

Also, anyone knows any example of a pre-trained convolutional net working properly in face recognition? An example could be a great help too

Thanks in advance, 

Sergi

**Comment by bhack** [comment]
@longjon The power of issues triaging please! :)

**Comment by flx42** [comment]
\*imitating @longjon\*
This tracker is for issues and bugs with Caffe. Please ask usage and installation questions on the [caffe-users](https://groups.google.com/forum/#!forum/caffe-users) list. Thanks!

@bhack it's CVPR right now, what did you expect? :)

**Comment by longjon** [comment]
Indeed. (Thanks @bhack @flx42, we're happy to have help with issue triaging in clear cases.)

**Comment by bhack** [comment]
@longjon Github permission are not so fine graded to add us only with labeling permission.
